Sat Jun 17 13:39:02 CEST 2006  VMiklos <vmiklos@frugalware.org>
  * libfwgrubconfig: log to FWGRUB_LOGDEV, not to stdout when installing to /dev/md*
diff -rN -u old-frugalwareutils/libfwgrubconfig/libfwgrubconfig.c new-frugalwareutils/libfwgrubconfig/libfwgrubconfig.c
--- old-frugalwareutils/libfwgrubconfig/libfwgrubconfig.c	2006-06-17 14:01:25.000000000 +0200
+++ new-frugalwareutils/libfwgrubconfig/libfwgrubconfig.c	2006-06-17 14:01:25.000000000 +0200
@@ -268,7 +268,8 @@
 			snprintf(dest, PATH_MAX, "/boot/grub/%s", ent->d_name);
 			fwutil_cp(src, dest);
 		}
-		pp = popen("grub --batch --device-map=/boot/grub/device.map", "w");
+		pp = popen("grub --batch --device-map=/boot/grub/device.map %s >"
+			FWGRUB_LOGDEV " 2>&1", "w");
 		if(!pp)
 			return(1);
 		fprintf(pp, "quit\n");
@@ -280,7 +281,8 @@
 			for(i=0;i<g_list_length(devs);i++)
 			{
 				char *dev = g_list_nth_data(devs, i);
-				pp = popen("grub --batch", "w");
+				pp = popen("grub --batch %s >" FWGRUB_LOGDEV
+					" 2>&1", "w");
 				if(!pp)
 					return(1);
 				fprintf(pp, "root %s\n", grub_convert(dev, 0));
@@ -296,7 +298,8 @@
 			for(i=0;i<g_list_length(devs);i++)
 			{
 				char *dev = g_list_nth_data(devs, i);
-				pp = popen("grub --batch", "w");
+				pp = popen("grub --batch %s >" FWGRUB_LOGDEV
+					" 2>&1", "w");
 				if(!pp)
 					return(1);
 				fprintf(pp, "root %s\n", grub_convert(dev, 0));

